numba/tests/test_pycc.py

```diff
@@ -233,6 +257,35 @@ def check_compile_for_cpu(self, cpu_name):
             self.assertPreciseEqual(res, 123 * 321)
             self.assertEqual(lib.multi.__module__, 'pycc_test_simple')
 
+    def check_gil_released(self, func):
+        for n_threads in (4, 12, 32):
+            # Try harder each time. On an empty machine 4 threads seems
+            # sufficient, but in some contexts (e.g. Travis CI) we need more.
+            arr = self.run_in_threads(func, n_threads)
+            distinct = set(arr)
+            try:
+                self.assertGreater(len(distinct), 1, distinct)
+            except AssertionError as e:
+                failure = e
+            else:
+                return
+        raise failure
+
+    def check_gil_held(self, func):
+        arr = self.run_in_threads(func, n_threads=4)
+        distinct = set(arr)
+        self.assertEqual(len(distinct), 1, distinct)
+
+    def test_cc_export_nogil(self):
+        with self.check_cc_compiled(self._test_module.cc) as lib:
+            self.assertEqual(lib.f_nogil.__module__, 'pycc_test_simple')
+            self.check_gil_released(lib.f_nogil)
```

Comment:
This test fails locally. However, the `nogil` flag works fine locally when used in `test_gil.py`. I tried removing the context manager but that changed nothing.

Reply:
It fails with `n_threads=32` in `self.check_gil_released`. For 4 and 12 it works fine.